lib/puppet/type/cloudwatch_alarm.rb

```diff
@@ -0,0 +1,91 @@
+Puppet::Type.newtype(:cloudwatch_alarm) do
+  @doc = 'type representing an AWS CloudWatch Alarm'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'the name of the alarm'
```

Comment:
Thanks, fixed for the autoscaling group types in a future commit. I'll change the other types as well separately to avoid merge issues.